### PR TITLE
feat: Allow VerifiedClient DataCap to be topped up

### LINKED ISSUE #1325: Clients who apply for additional DataCap need to use a new address every time
In the current implementation of spec-actors we require Clients to treat DataCap as a "one time" credit card - per address you can only receive DataCap once. 

Practically, this introduces complexity for normal clients, who as a result need to manage which address of theres has DataCap, which addresses have already _had_ DataCap, and so on. 

Especially for web3 apps (like Slate) where the end-user is in control of their keys (but enabling Slate to handle the deal making) - we run into a scenario where the web3 dev needs to manage swapping out addresses for DataCap allocation / deal making unnecessarily.

### PR DESCRIPTION
**Pull request by placer14**
Implements [FIP-0012](https://github.com/filecoin-project/FIPs/blob/5ab54f271a87b3c759c21dc36948c033f105ac4d/FIPS/fip-0012.md) and fixes #1325.

Thoughts on these changes:
- The mock runtime `addVerifiedClient` function is now overloaded to support an optional `expectedAllowance` arg to be used in the cases where VerifiedClients will now have more than what was initially allotted. This makes the verification of an `expectedAllownace` a concern on the caller rather than letting the function reflect on its internal before/after state to verify correctness. I'd personally prefer the latter to keep the call simple, but there were errors when I attempted to access the runtime map for the "before" state of the Client's DataCap allowance. (An example of that is at https://gist.github.com/placer14/f2134a4064373607da44b48b4f338d69.)
- This change will mix all Verifier's allowances together for each Client address. If there were ever a need to know how DataCap is used on a per-Verifier basis, we would be eliminating that option here.
- The unit test which protects that a VerifiedClient can be referred to be its ID and non-ID addr is also protecting its ability to be topped-up. If it's preferred that these behaviors are protected by separate tests, I can, but felt the redundancy was unnecessary and left them combined.
- This change does not limit the DataCap upper bound and expects that DataCap boundaries are already protected where necessary.

### CODE REVIEW

File: actors/builtin/verifreg/verified_registry_test.go
Comment (anorth) [inline actors/builtin/verifreg/verified_registry_test.go:813]: Fail the test rather than ignore an unexpected argument. Fail also if the slice has length zero.

Alternatively, I would be just as happy if the two args were explicit, and the usually-redundant expected cap was always explicitly passed.

**Review by anorth — APPROVED** [review]
Looks good to me, but please wait for approval from @ZenGround0

File: actors/builtin/verifreg/verified_registry_test.go
Comment (ZenGround0) [reply actors/builtin/verifreg/verified_registry_test.go:813]: I want to change this to two explict arguments.

**Review by ZenGround0 — COMMENTED** [review]
Since your week of work is over @placer14 I'm taking this PR over.  I'm going to make some changes to testing but the code change looks good.

**Review by ZenGround0 — APPROVED** [review]

**Pull request merged**
